### DISCUSSION TITLE
Is it possible to use EC for all data written, as it's written?

### DISCUSSION THREAD

**Discussion by danthegoodman1** [seaweedfs/seaweedfs#5919] [Q&A] [answered]
I've been reading through the EC wiki and am wondering if it's possible (or advised) to use EC for "all data" in an S3 bucket.

I know that files would be 100% immutable (never updated), so it seems like that aleviates some of the drawbacks of EC (lacking updates).

However, it seems like SWFS starts with normal replicated volumes, then turns them into EC, rather than starting with files as EC like something such as minio might.

Am I right in understanding that this is not really possible?

I ask because I'm toying with the idea of building a 100TB-1PB dataset and would much prefer to have 1.4x storage rather than 3x to replicate!

**Comment by chrislusf** [comment, marked as answer]
If you encode the volumes fast enough, the replicated volumes will just become buffers.

You can also reduce or avoid replication before EC.

**Reply by danthegoodman1** [reply]
Thanks, but if you use no replication before EC then isn’t there a brief period where the data is considered durable (eg s3 api responded to put) but non replicated?

**Reply by chrislusf** [reply]
Yes
